### DISCUSSION TITLE
ldap

### DISCUSSION THREAD

**Discussion by szaimen** [nextcloud/all-in-one#406] [Ideas] [answered]
as option
but without automatic configuration for the start (should be documented then)
but could automatically install needed apps, etc.
https://hub.docker.com/r/bitnami/openldap
Or
https://github.com/osixia/docker-openldap

I guess first does the image need to be updated

**Comment by ghost** [comment]
Yes I personally find good, but would prefer the automatic configuration.
#144

**Comment by szaimen** [comment]
I fear this does not pass all requirements for integrating this new container into AIO.
See https://github.com/nextcloud/all-in-one#requirements-for-integrating-new-containers

**Reply by szaimen** [reply]
However you might be interested in this: #1581

**Comment by szaimen** [comment]
Hi, I just wanted to mention that we have this now: https://github.com/nextcloud/all-in-one/tree/main/community-containers#how-to-add-containers. So the community could potentially add e.g. https://hub.docker.com/r/nitnelave/lldap as additional container. Feel free to ping me if you should need help on this!

**Reply by szaimen** [reply]
Regarding the json, I just leave this here as an example:

```json
{
    "aio_services_v1": [
        {
            "container_name": "nextcloud-aio-lldap",
            "display_name": "Light LDAP implementation",
            "documentation": "https://github.com/nextcloud/all-in-one/tree/main/community-containers/lldap",
            "image": "lldap/lldap",
            "image_tag": "v0",
            "internal_port": "17170",
            "restart": "unless-stopped",
            "ports": [
                {
                  "ip_binding": "%APACHE_IP_BINDING%",
                  "port_number": "17170",
                  "protocol": "tcp"
                }
            ],
            "apparmor_unconfined": true,
            "environment": [
                "TZ=%TIMEZONE%",
                "UID=65534",
                "GID=65534",
                "LLDAP_JWT_SECRET=%LLDAP_JWT_SECRET%",
                "LLDAP_LDAP_USER_PASS=%LLDAP_LDAP_USER_PASS%",
                "LLDAP_LDAP_BASE_DN=%NC_DOMAIN_BASE_DN%", // needs adjustments in AIO to return the correct one
            ],
            "secrets": [
                "LLDAP_JWT_SECRET",
                "LLDAP_LDAP_USER_PASS"
            ]
            "volumes": [
                {
                    "source": "nextcloud_aio_lldap",
                    "destination": "/data",
                    "writeable": true
                }
            ],
            "backup_volumes": [
                "nextcloud_aio_lldap"
            ]
        }
    ]
}
```

**Reply by szaimen** [reply]
@FlattusBlastus since you are very interested in this festure IIRC, would you be up for creating the PR that adds this?

**Reply by FlattusBlastus** [reply]
lldap seems pretty straight forward. I'll be happy to dig in during the evenings this week.

**Reply by FlattusBlastus** [reply]
I expect to send my PR this weekend.

**Reply by szaimen** [reply]
Hi @FlattusBlastus any update here?

**Comment by szaimen** [comment, marked as answer]
This was done in https://github.com/nextcloud/all-in-one/pull/4398

**Reply by szaimen** [reply]
This is now released with v8.2.0 Beta. Testing and feedback is welcome! See https://github.com/nextcloud/all-in-one#how-to-switch-the-channel

**Reply by kri164** [reply]
Are there any alternative container registry as Github or opensuse?  Reaching limit every day.

```
nextcloud-aio-mastercontainer[29394]: Trying to pull docker.io/nextcloud/all-in-one:beta...
nextcloud-aio-mastercontainer[29394]: Pulling image //nextcloud/all-in-one:beta inside systemd: setting pull timeout to 5m0s
nextcloud-aio-mastercontainer[29394]: time="2024-04-16T13:10:56+02:00" level=warning msg="Failed, retrying in 1s ... (1/3). Error: initializing source docker://nextcloud/all-in-one:beta: reading manifest beta in docker.io/nextcloud/all-in-one: toomanyrequests: You have reached your pull rate limit. You may increase 
the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit"

```
